### PR TITLE
Provide debug only link-out to CRM

### PR DESCRIPTION
**Pull request by allthesignals**
This PR adds a query param to the "show project" route, `debug`, which is false by default.

This query param is used to show or hide a hyperlink that constructs a URL for the related CRM project page.

image

I'm add this to make it easier to get to the CRM information, which is the source of truth. This should make debugging a little easier.

### CODE REVIEW

File: client/app/models/project.js [hunk 1]
```diff
@@ -65,6 +65,8 @@ export default class ProjectModel extends Model {
 
   @attr('string') dcpProjectname;
 
+  @attr('string') dcpProjectid;
```
Comment (godfreyyeung) [inline client/app/models/project.js:68]: In terms of process, these should be in a separate commit, instead of hidden in a commit indicating something completely different

File: client/app/templates/show-project.hbs [hunk 2]
```diff
@@ -8,8 +8,13 @@
           {{else}}
             Project {{model.dcpName}}
           {{/if}}
-        </h1>
 
+          {{#if this.debug}}
+            <a href="https://nycdcppfs.crm9.dynamics.com/main.aspx?etn=dcp_project&id={{{model.dcpProjectid}}}&pagetype=entityrecord" target="_blank">
```
Comment (godfreyyeung) [inline client/app/templates/show-project.hbs:13]: This should be dependent on environment. We're usually developing locally, and working with data on UAT2. UAT2 data can be pretty different from prod data

**Review by godfreyyeung — CHANGES_REQUESTED** [review]

File: client/app/models/project.js (hunk 1 above)
Comment (allthesignals) [reply client/app/models/project.js:68]: > In terms of process, these should be in a separate commit, instead of hidden in a commit indicating something completely different

I'm sorry — what do you mean by this? It's already being shipped with the response, and used in a template. How does the commit "indicate something completely different"?

File: client/app/templates/show-project.hbs (hunk 2 above)
Comment (allthesignals) [reply client/app/templates/show-project.hbs:13]: > This should be dependent on environment. We're ~usually~ always developing locally, and working with data on UAT2. UAT2 data can be pretty different from prod data

Sure, I can add that.

File: client/app/models/project.js (hunk 1 above)
Comment (allthesignals) [reply client/app/models/project.js:68]: Confused about why you think this is some kind of process violation — it's a change that's necessary for this commit to function at all.

Comment (godfreyyeung) [reply client/app/models/project.js:68]: Oh sorry, the green over the dcpFemafloodzoneshadedx made me think multiple attributes were being added. 
In this case, why can't `project.id` or `project.dcpName`  be used, instead of dcpProjectid?

Comment (allthesignals) [reply client/app/models/project.js:68]: > Oh sorry, the green over the dcpFemafloodzoneshadedx made me think multiple attributes were being added.
> In this case, why can't `project.id` or `project.dcpName` be used, instead of dcpProjectid?

Because this model was set up to identify projects by their dcpName values, not the regular CRM ids.

We can't use dcpName because you can't link out to CRM projects with the dcpName value. It requires the GUID.

Comment (godfreyyeung) [reply client/app/models/project.js:68]: Ah okay, gotcha, thank you

**Review by godfreyyeung — APPROVED** [review]

**Review by trbmcginnis — APPROVED** [review]
